bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/handler/CaddxBaseThingHandler.java

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.caddx.internal.handler;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.caddx.internal.CaddxEvent;
+import org.openhab.binding.caddx.internal.config.CaddxKeypadConfiguration;
+import org.openhab.binding.caddx.internal.config.CaddxPartitionConfiguration;
+import org.openhab.binding.caddx.internal.config.CaddxZoneConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract class for a Caddx Thing Handler.
+ *
+ * @author Georgios Moutsos - Initial contribution
+ */
+@NonNullByDefault
+public abstract class CaddxBaseThingHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(CaddxBaseThingHandler.class);
+
+    /** Bridge Handler for the Thing. */
+    private @Nullable CaddxBridgeHandler caddxBridgeHandler = null;
+
+    /** Caddx Alarm Thing type. */
+    private CaddxThingType caddxThingType;// = null;
+
+    /** Caddx Properties. */
+    private boolean thingHandlerInitialized = false;
+
+    /** Partition Number. */
+    private int partitionNumber;
+
+    /** Zone Number. */
+    private int zoneNumber;
+
+    /** Keypad Address. */
+    private int keypadAddress;
+
+    public CaddxBaseThingHandler(Thing thing, CaddxThingType caddxThingType) {
+        super(thing);
+        this.caddxThingType = caddxThingType;
+    }
+
+    @Override
+    public void initialize() {
+        logger.trace("initialize(): {}; Thing ID: {}.", caddxThingType, this.getThing().getUID());
+
+        getConfiguration(caddxThingType);
+
+        // set the Thing offline for now
+        updateStatus(ThingStatus.OFFLINE);
```

Comment:
How does this become online?

Reply:
This is the base class. 
The things change status to online in the child classes. [ThingHandlerKeypad, ThingHandlerPanel, ThingHandlerPartition, ThingHandlerZone].